README.md

```diff
@@ -1 +1,122 @@
+* `result_buffers` controls in which format the results are stored. These result buffers hold results from `populate_result_buffer!` and are passed to `consume!`. This allows the user to have multiple result formats for the with `parsing_ctx` e.g. row oriented vs column oriented buffers.
+
+See the docstring of `populate_result_buffer!` and `consume!` for more information about how to integrate with them.
+
+## Example: Examining the results of the Lexer
+
+Since `ChunkedBase.jl` handles newline detection automatically, a very simple chunked processor could just show the newline positions to the user. Of course, this is not very useful apart from demonstrating how to use this package.
+
+```julia
+using ChunkedBase
+using NewlineLexers: Lexer
+
+# Our chunked processor doesn't require any settings nor does it need to maintain
+# any additional state we'll define our `ParsingContext` and `ConsumeContext` only
+# for dispatch reasons.
+struct ParsingContext <: AbstractParsingContext end
+struct ConsumeContext <: AbstractConsumeContext end
+# Our buffer will hold the newline positions in the current chunk
+struct ResultBuffer <: AbstractResultBuffer
+    newlines_in_chunk::Vector{Int}
+end
+
+# Our overload copies `newlines_segment` to our result buffer
+function ChunkedBase.populate_result_buffer!(
+    result_buf::ResultBuffer,
+    newlines_segment::AbstractVector{Int32},
+    ::ParsingContext,
+    ::Vector{UInt8},
+    ::Union{Nothing,Vector{UInt8}}=nothing,
+    ::Type=Tuple{}
+)
+    resize!(result_buf.newlines_in_chunk, length(newlines_segment))
+    result_buf.newlines_in_chunk .= copy(newlines_segment)
+    return nothing
+end
+
+# We consume result buffer by simply printing it
+function ChunkedBase.consume!(::ConsumeContext, payload::ParsedPayload)
+    chunking_ctx = payload.chunking_ctx
+    result_buffer = payload.results
+
+    # To demonstrate how double-buffering works, print the buffer id and the refill number
+    chunk_info = (chunking_ctx.id, chunking_ctx.buffer_refills[])
+    @info "Newlines in chunk (id:$chunk_info): $(result_buffer.newlines_in_chunk)"
+    return nothing
+end
+```
+
+Now that we defined all overloads and contexts we can define our user-facing function:
+
+```julia
+# Prints relative positions of `\n` in each ingested chunk of size `buffersize`,
+# using `nworkers` spawned tasks.
+function print_newlines(io, buffersize, nworkers)
+    lexer = Lexer(io, nothing, '\n')
+    parsing_ctx = ParsingContext()
+    consume_ctx = ConsumeContext()
+    chunking_ctx = ChunkingContext(buffersize, nworkers, 0, nothing)
+    # ChunkedBase.jl requires 2 result buffers per worker task, we'd get an error otherwise
+    result_buffers = [ResultBuffer(Int[]) for _ in 1:2nworkers]
```

Comment:
Would it make sense to put this behind a helper function, like
```julia
    result_buffers = init_result_buffers(ResultBuffer, chunking_ctx)
```
?
It seems strange to ask the users to init these buffers according to a specific pattern that is internal to this package. I think it would be better to encapsulate that.
(This kind of API clean up seems in-scope to me for a documentation / understandability / maintainability project - it's not about adding new features or major refactoring, it's about cleaning up the interface, which feels like part of improving the documentation.)

Reply:
Oh, i guess this would require ResultBuffers to have a default constructor. That seems reasonable to me? Or you could have them pass in a factory function like `init_result_buffers(()->ResultBuffer(Int[]), chunking_ctx)`?

Reply:
I'd prefer not to increase the API surface area... This is documented, trivial to satisfy and this whole package is targetting package authors. In ChunkedCSV.jl we have our own `_make_result_buffers` helper that tries to be clever about minizing allocations so we wouldn't even want to use the generic function.

Reply:
Cool, thanks, makes sense.